compiler/backend/src/org/jetbrains/kotlin/codegen/state/KotlinTypeMapper.kt

```diff
@@ -1617,7 +1621,8 @@ class KotlinTypeMapper @JvmOverloads constructor(
             val containingDeclaration = descriptor.containingDeclaration
             return containingDeclaration.isInlineClass() &&
                     descriptor.kind == CallableMemberDescriptor.Kind.SYNTHESIZED &&
-                    descriptor.name == InlineClassDescriptorResolver.BOX_METHOD_NAME
+                    (descriptor.name == InlineClassDescriptorResolver.BOX_METHOD_NAME ||
+                     descriptor.name.asString() == BOX_JVM_METHOD_NAME)
```

Comment:
Is it possible to add test for old backend in this case?

Reply:
Nice catch. Done